pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -258,6 +258,8 @@ def setup_distributed(self):
         self.dist.rank = self.global_rank
         self.dist.device = self.root_device
 
+        trainer._call_setup_datamodule_hook()
```

Comment:
@tchaton could you describe why calling `datamodule.setup` in the trainer after `setup_environment` doesn't work? i'm concerned that having the lightning module and datamodule setups being called in two very different places will lead to other bugs down the line